game-core/src/main/java/games/strategy/triplea/ui/SelectTerritoryDialog.java

```diff
@@ -25,7 +26,8 @@
   SelectTerritoryDialog(final Frame owner, final String title, final Collection<Territory> territories) {
     super(owner, title, true);
 
-    territoryComboBox = new JComboBox<>(territories.toArray(new Territory[0]));
+    territoryComboBox = new JComboBox<>(territories.stream().sorted().toArray(Territory[]::new));
+    AutoCompletion.enable(territoryComboBox);
```

Comment:
Quick run-by comment, @ssoloff did you consider adding an auto-complete property to a Jcombo box builder? This would avoid the static call FWIW, eg:
```
JComboBox.builder()
  .values(territories..)
  .enableAutoComplete()
```

Reply:
I did not consider using `JComboBoxBuilder` at all because it is hard-coded to use `String` elements.  However, your suggestion would encapsulate the `AutoCompletion` class (or whatever we ultimately use) nicely.  There's currently no other clients of `JComboBoxBuilder`, so I'll see if it's not too bad to generify it.

Reply:
:grin:  'hard-coded'; in one respect I fully agree. In another, it's intended to be a simplification where 95% of jcombo boxes display strings. With luck that assumption leads to things being far simpler. If that does not hold, my intent was for us to use the swing jcombo box directly or see about adding an API that would fit in with the builder (and provide type safety)

Reply:
3cee3b5fc makes the necessary changes to use `JComboBoxBuilder` and hides `AutoCompletion` usage within the builder, as suggested.

I removed some unused functionality related to `ClientSetting`.  Otherwise it would require introducing new abstractions related to converting generic types to/from strings, as required by the `ClientSetting` subsystem.  Since no one currently makes use of this feature, I punted and just removed it. :smile:

I also renamed some members for consistency.

Happy to revert if it's not the way you envisioned `JComboBoxBuilder` evolving.  Note that for the 95% strings case, we could always add a `JComboBoxBuilder#builder()` overload that is simply:

```java
public static JComboBoxBuilder<String> builder() {
  return builder(String.class);
}
```